packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx

```diff
@@ -350,15 +363,261 @@ export function preRenderLayout(
+      sourceToTarget = {};
+      result[sourceId] = sourceToTarget;
+    }
+    sourceToTarget[idFromNumberOrSNode(link.target)] = { reactId: getId('link') };
+  });
+
+  return result;
+}
+
+type ItemPositions = {
+  readonly x0: number;
+  readonly y0: number;
+  readonly x1: number;
+  readonly y1: number;
+};
+
+type SankeyLinkWithPositions = ItemPositions & {
+  readonly source: ItemPositions;
+  readonly target: ItemPositions;
+  readonly width: number;
+};
+
+type AreaDataPoint = {
+  readonly x: number;
+  readonly y0: number;
+  readonly y1: number;
+};
+
+const linkToDataPoints = (d: SankeyLinkWithPositions): [AreaDataPoint, AreaDataPoint] => {
+  const halfWidth = d.width * 0.5;
+  return [
+    { x: d.source.x1, y0: d.y0 + halfWidth, y1: d.y0 - halfWidth },
+    { x: d.target.x0, y0: d.y1 + halfWidth, y1: d.y1 - halfWidth },
+  ];
+};
+
+const linkArea = d3Area()
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  .x((p: any) => p.x)
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  .y0((p: any) => p.y0)
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  .y1((p: any) => p.y1)
+  .curve(d3CurveBasis);
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type TooltipDiv = D3Selection<BaseType, unknown, HTMLElement, any>;
+
+function nodeTextColor(state: Readonly<ISankeyChartState>, singleNode: SNode): string {
+  return !(
+    !state.selectedState ||
+    (state.selectedState && state.selectedNodes.has(singleNode.index!) && state.selectedNode) ||
+    (state.selectedState && !state.selectedNode)
+  )
+    ? DEFAULT_TEXT_COLOR
+    : NON_SELECTED_TEXT_COLOR;
+}
+
+// NOTE: To start employing React.useMemo properly, we need to convert this code from a React.Component
+// to a function component. This will require a significant refactor of the code in this file.
+// https://stackoverflow.com/questions/60223362/fast-way-to-convert-react-class-component-to-functional-component
+// I am concerned that doing so would break this contract, making it difficult for consuming code.
 export class SankeyChartBase extends React.Component<ISankeyChartProps, ISankeyChartState> {
   private chartContainer: HTMLDivElement;
   private _reqID: number;
-  private _calloutId: string;
-  private _linkId: string;
-  private _margins: IMargins;
-  private _isRtl: boolean = getRTL();
-  private _normalizeData: NormalizeFunction;
-  private _emptyChartId: string;
+  private readonly _calloutId: string;
+  private readonly _linkId: string;
+  private readonly _chartId: string;
+  private readonly _emptyChartId: string;
+  private readonly _labelTooltipId: string;
+  private readonly _margins: IMargins;
+  private readonly _isRtl: boolean = getRTL();
+
+  private readonly _computeClassNamesProps: (
+    theme: ITheme,
+    pathColor: string,
+    className: string,
+    containerWidth: number,
+    containerHeight: number,
+  ) => ISankeyChartStyleProps;
+  private readonly _computeClassNames: (
```

Comment:
>_computeClassNames [](http://example.com/codeflow?start=19&length=18)

this can be achieved by adding the tooltipDiv style to the .style.tsx file where the style prop linkage is already setup